### PR TITLE
feat: add support for nginx reverse proxy

### PR DESCRIPTION
**Pull request by wollefitz**
Adds the possibility to start with an nginx reverse proxy in front of Zeebe which handles TLS termination

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -11,6 +26,9 @@ services:
       - ZEEBE_BROKER_GATEWAY_SECURITY_CERTIFICATECHAINPATH=/usr/local/zeebe/server.crt
       - ZEEBE_BROKER_GATEWAY_SECURITY_PRIVATEKEYPATH=/usr/local/zeebe/server.pem
       - ZEEBE_BROKER_GATEWAY_NETWORK_PORT=${ZEEBE_PORT:-26500}
+      - VIRTUAL_HOST=${ZEEBE_HOSTNAME:-sub.example.com}
```
Comment (nikku) [inline docker-compose.yml:29]: Why do we add these? Is this special configuration provided to `zeebe`?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,6 +1,21 @@
 version: "2"
 
 services:
+  nginx:
```
Comment (nikku) [inline docker-compose.yml:4]: Could you elaborate how forwarding works here? I.e. how do requests to `test.example.com:NGINX_PORT` end up in `test.example.com:ZEEBE_PORT`.

File: docker-compose.yml (hunk 1 above)
Comment (wollefitz) [reply docker-compose.yml:29]: These are consumed by the nginx proxy, see https://github.com/nginx-proxy/nginx-proxy#usage. I'll add a comment explaining the concept.

File: README.md [hunk 3]
```diff
@@ -91,6 +92,9 @@ ZEEBE_HOSTNAME=sub.example.com sh -c 'echo "127.0.0.1    $ZEEBE_HOSTNAME"' | sud
 # start zeebe with security enabled
 ZEEBE_HOSTNAME=sub.example.com docker-compose up zeebe
 
+# start zeebe with  nginx reverse proxy terminating TLS and security disabled
+ZEEBE_HOSTNAME=sub.example.com ZEEBE_BROKER_GATEWAY_SECURITY_ENABLED=false docker-compose up zeebe nginx
```
Comment (wollefitz) [inline README.md:96]: I just now realized that I forgot to mention naming conventions of the certs in order to be picked up automatically by the nginx proxy, will add this.

File: docker-compose.yml (hunk 1 above)
Comment (nikku) [reply docker-compose.yml:29]: What would a non-magic alternative look like? I guess simply configuring the [nginx standalone](https://hub.docker.com/_/nginx/) (so we know what is happening under the hood?).

**Comment by nikku** [comment]
@wollefitz Thanks again for your PR. Based on your contribution I believe https://github.com/camunda/zeebe-connection-test/commit/b527c483b78c667e453905af49191023c1be449d and https://github.com/camunda/zeebe-connection-test/commit/6cbc0af3af79309ab909ab9c46e3b0867226ef62 accomplish the same thing, with a little less magic. 

Please have a look!

File: README.md (hunk 3 above)
Comment (nikku) [reply README.md:96]: I think I understood the setup now, and added a working CI test, too, cf. https://github.com/camunda/zeebe-connection-test/pull/6#issuecomment-1520376191.

**Pull request closed without merging**